### PR TITLE
Csharp docfx update

### PR DESCRIPTION
**Pull request by cassiebreviu**
**Description**: update csharp docs to create in a folder named "csharp". Update CI and gitignore to remove unused csharp files to fix CI on gh-pages.

### CODE REVIEW

File: csharp/ApiDocs/.gitignore [hunk 1]
```diff
@@ -7,4 +7,6 @@
 /**/bin/
 /**/obj/
 _site
-dest
\ No newline at end of file
+dest
+csharp
+*.txt
```
Comment (yuslepukhin) [inline csharp/ApiDocs/.gitignore:12]: Is this for something that gets generated? Can you reply in the comments for the record, what specifically. Thanks!

File: .github/workflows/publish-csharp-apidocs.yml [hunk 2]
```diff
@@ -26,11 +26,12 @@ jobs:
         clean: false
     - name: Move API docs into target area
       run: |
-        cd docs
-        cd api
-        mkdir csharp
-        cd ../../
-        MOVE csharp/ApiDocs/dest docs/api/csharp
+        rm -r -fo docs/api/csharp
```
Comment (yuslepukhin) [inline .github/workflows/publish-csharp-apidocs.yml:29]: >rm -r -fo docs/api/csharp [](http://example.com/codeflow?start=8&length=25)

On windows rm is not there. Also CMD would not recognize forward slashes.

Comment (cassiebreviu) [reply .github/workflows/publish-csharp-apidocs.yml:29]: So I had `del` for `cmd` but it didnt work in the gh action, I had to update to the powershell commands to get it to work.

File: csharp/ApiDocs/.gitignore (hunk 1 above)
Comment (cassiebreviu) [reply csharp/ApiDocs/.gitignore:12]: Yes, these get generated when you build the project. When you build docfx it generates an out folder with the static site. The folder name is specified in the docfx.json file.

File: .github/workflows/publish-csharp-apidocs.yml (hunk 2 above)
Comment (cassiebreviu) [reply .github/workflows/publish-csharp-apidocs.yml:29]: yea `fo` is `force` same as `f` I believe.

**Review by yuslepukhin — APPROVED** [review]

**Pull request merged**